### DISCUSSION TITLE
Problem trying to fork CC

### DISCUSSION THREAD

**Discussion by NevesAngelesLeo** [cc-tweaked/CC-Tweaked#1800] [Q&A] [answered]
i've been following https://github.com/cc-tweaked/CC-Tweaked/blob/mc-1.20.x/CONTRIBUTING.md#developing tutorial on how to start forking CC for an CC mod project im working with my friends. When it comes to the part i need to run ./gradlew build, it always fails one or more tests on core. Can someone tell me how do i fix this? Is something on my machine? Any help is appreciated :)
![image](https://github.com/cc-tweaked/CC-Tweaked/assets/119637072/f498fa9c-e6c8-4fd0-849e-6f88c9f5d3d4)

**Comment by SquidDev** [comment]
Oh, that's very odd, especially given it sounds like that happens every time. If you *just* run `./gradlew :core:test`, does that fare any better?

**Comment by NevesAngelesLeo** [comment]
It failed again, this time only once 

![image](https://github.com/cc-tweaked/CC-Tweaked/assets/119637072/39feae3c-da26-4676-bb22-346b999c16b4)

![image](https://github.com/cc-tweaked/CC-Tweaked/assets/119637072/59ee0cb8-e94c-45e3-9b11-9459350ddb32)
 
The standard output was this:

Computer is currently on.
 1 | 1 GPS requests served                                                            |
 2 | 1 GPS requests served                                                            |
 3 | 1 GPS requests served                                                            |
 4 | 1 GPS requests served                                                            |
 5 | 1 GPS requests served                                                            |
 6 | ?Position is 5,5,5                                                               |
 7 | Opening channel on modem back                                                    |
 8 | Position is 10,5,5                                                               |
 9 | Opening channel on modem back                                                    |
10 | Position is 5,10,5                                                               |
11 | Opening channel on modem back                                                    |
12 | 1 GPS requests served                                                            |
13 | 1 GPS requests served                                                            |
14 | 1 GPS requests served                                                            |
15 | ????????????????????????????????????º????????????º?????????????????????????????? |
16 | ??º???????????º??º???????????????????????????????????º????????????º????????????? |
17 | ????????º????????????????????º???????????????????????????????º?????????????????? |
18 | ??????????????????????????????????º????????????????????????????????????????????? |
19 | ???????????????????????????????????????????????????????????????????????????????? |
20 | ???????????????????????????????????????????????????????????????????????????????? |
21 | ???????????????????????????????????????????????????????????????????????????????? |
22 | ??????????????????????????????????????????????????????????????1 modem found.     |
23 | 1 message repeated.                                                              |
24 | ?1 modem found.                                                                  |
25 | 0 messages repeated.                                                             |
26 | 1 modem found.                                                                   |
27 | 1 message repeated.                                                              |
28 | ?1 modem found.                                                                  |
29 | 2 messages repeated.                                                             |
30 | ???1 modem found.                                                                |
31 | 2 messages repeated.                                                             |
32 | ???????????????????????????????????????????????????????????????????????????????? |
33 | ??????????????????Cannot use the shell as a hashbang program                     |
34 | ?????Hashbang recursion depth limit reached when loading file:                   |
35 | test-files/out.lua                                                               |
36 | ???????????????????????????????????????????????????????????????????????????????? |
37 | ???????????????????????????????????????????????????????????????????????????????? |
38 | ???????????????????????????????????????????????????????????????????????????????? |
39 | ???????????????????????????????????????????????????????????????????????????????? |
40 | ???????????????????????????????????????????????????????????????????????????????? |
41 | ???????????????????????????????????????????????????????????????????????????????? |
42 | ???????????????????????????????????????????????????????????????????????????????? |
43 | ?                                                                                |
44 |                                                                                  |
45 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
46 | nst/JSONTestSuite ? n_structure_open_array_object                                |
47 |   ...pis/textutils_spec.lua:277                                                  |
48 |                                                                                  |
49 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
50 | nst/JSONTestSuite ? n_structure_100000_opening_arrays                            |
51 |   ...pis/textutils_spec.lua:277                                                  |
52 |                                                                                  |
53 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
54 | nst/JSONTestSuite ? n_number_-2.                                                 |
55 |   ...pis/textutils_spec.lua:277                                                  |
56 |                                                                                  |
57 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
58 | nst/JSONTestSuite ? n_number_with_leading_zero                                   |
59 |   ...pis/textutils_spec.lua:277                                                  |
60 |                                                                                  |
61 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
62 | nst/JSONTestSuite ? n_number_real_without_fractional_part                        |
63 |   ...pis/textutils_spec.lua:277                                                  |
64 |                                                                                  |
65 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
66 | nst/JSONTestSuite ? n_number_neg_int_starting_with_zero                          |
67 |   ...pis/textutils_spec.lua:277                                                  |
68 |                                                                                  |
69 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
70 | nst/JSONTestSuite ? n_number_-01                                                 |
71 |   ...pis/textutils_spec.lua:277                                                  |
72 |                                                                                  |
73 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
74 | nst/JSONTestSuite ? n_number_0.e1                                                |
75 |   ...pis/textutils_spec.lua:277                                                  |
76 |                                                                                  |
77 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
78 | nst/JSONTestSuite ? n_number_2.e+3                                               |
79 |   ...pis/textutils_spec.lua:277                                                  |
80 |                                                                                  |
81 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
82 | nst/JSONTestSuite ? n_number_2.e-3                                               |
83 |   ...pis/textutils_spec.lua:277                                                  |
84 |                                                                                  |
85 | Pending ? The textutils library ? textutils.unserializeJSON ? passes             |
86 | nst/JSONTestSuite ? n_number_2.e3                                                |
87 |   ...pis/textutils_spec.lua:277                                                  |
88 |                                                                                  |
89 | Error ? cc.internal.event ? discard_char ? handles an empty event queue          |
90 |   ...nternal/event_spec.lua:21: Timeout                                          |
91 |   stack traceback:                                                               |
92 |   	[C]: in function 'error'                                                      |
93 |   	.../spec/test_helpers.lua:82: in upvalue 'timeout'                            |
94 |   	...nternal/event_spec.lua:21: in upvalue 'test'                               |
95 |   	...nternal/event_spec.lua:40: in function <...nternal/event_spec.lua:39>      |
96 |   	[C]: in function 'xpcall'                                                     |
97 |   	/test-rom/mcfly.lua:143                                                       |
98 |                                                                                  |
99 | Ran 1306 test(s), of which 1305 passed (99.9234%). Skipped 11 pending test(s).   |
100 | >

**Reply by SquidDev** [reply]
Hrm, I've no clue why those still fail! You can just skip running tests by doing `./gradlew assemble` rather than `./gradlew build` then.

**Reply by NevesAngelesLeo** [reply]
No luck, now i got this 

![image](https://github.com/cc-tweaked/CC-Tweaked/assets/119637072/dfb7e074-5c8f-4b44-932f-2f502f4fa4a0)

**Reply by SquidDev** [reply]
Can you post the full log? There should be some more output above that.

**Reply by NevesAngelesLeo** [reply]
There's not much, the output it gives in the terminal while trying to assemble gets deleted, if you can tell me an way to get more detais you can tell me

That's what i got

O Windows PowerShell
Copyright (C) Microsoft Corporation. Todos os direitos reservados.

Instale o PowerShell mais recente para obter novos recursos e aprimoramentos! https://aka.ms/PSWindows

PS C:\Users\Thiago\IdeaProjects\CCFork> cd CC-Tweaked
PS C:\Users\Thiago\IdeaProjects\CCFork\CC-Tweaked> ./gradlew assemble
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Configure project :fabric
Fabric Loom: 1.6.7

> Configure project :fabric-api
Fabric Loom: 1.6.7

> Task :web:npmInstall FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':web:npmInstall'.
> A problem occurred starting process 'command 'npm''

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.    

BUILD FAILED in 1m 17s
46 actionable tasks: 9 executed, 37 up-to-date
PS C:\Users\Thiago\IdeaProjects\CCFork\CC-Tweaked>

**Reply by SquidDev** [reply]
That's a slightly different error ( the npmInstall task is failing, not compileTeaVM), but this should be easy to fix - you'll need NodeJS/npm on the path. Sorry, the CONTRIBUTING.md docs are wrong there, since the multiloader refactor, we always build the docs now.

**Reply by NevesAngelesLeo** [reply]
Now i got this, something with my jdk? for one sec i believed that we did it XD

PS C:\Users\Thiago\IdeaProjects\CCFork> cd CC-Tweaked
PS C:\Users\Thiago\IdeaProjects\CCFork\CC-Tweaked> ./gradlew assemble 
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details

> Configure project :fabric
Fabric Loom: 1.6.7

> Configure project :fabric-api
Fabric Loom: 1.6.7

> Task :web:compileTeaVM
Replacing dan200/computercraft/core/apis/http/request/HttpRequest with dan200/computercraft/core/apis/http/request/THttpRequest
Replacing dan200/computercraft/core/apis/http/CheckUrl with dan200/computercraft/core/apis/http/TCheckUrl
Replacing dan200/computercraft/core/apis/http/websocket/Websocket with dan200/computercraft/core/apis/http/websocket/TWebsocket
Replacing dan200/computercraft/core/asm/LuaMethodSupplier with dan200/computercraft/core/asm/TLuaMethodSupplier
Replacing dan200/computercraft/core/asm/PeripheralMethodSupplier with dan200/computercraft/core/asm/TPeripheralMethodSupplier
Replacing dan200/computercraft/core/computer/computerthread/ComputerThread$ExecutorImpl with dan200/computercraft/core/computer/computerthread/TComputerThread$ExecutorImpl
Replacing dan200/computercraft/core/computer/computerthread/ComputerThread$TimeoutImpl with dan200/computercraft/core/computer/computerthread/TComputerThread$TimeoutImpl
Replacing dan200/computercraft/core/computer/computerthread/ComputerThread with dan200/computercraft/core/computer/computerthread/TComputerThread
Replacing dan200/computercraft/core/CoreConfig with dan200/computercraft/core/TCoreConfig
Replacing io/netty/handler/codec/http/DefaultHttpHeaders with io/netty/handler/codec/http/TDefaultHttpHeaders
Replacing io/netty/util/AsciiString with io/netty/util/TAsciiString
Replacing org/slf4j/LoggerFactory$1 with org/slf4j/TLoggerFactory$1
Replacing org/slf4j/LoggerFactory$LoggerImpl with org/slf4j/TLoggerFactory$LoggerImpl
Replacing org/slf4j/LoggerFactory with org/slf4j/TLoggerFactory
Replacing org/slf4j/MarkerFactory$MarkerImpl with org/slf4j/TMarkerFactory$MarkerImpl
Replacing org/slf4j/MarkerFactory with org/slf4j/TMarkerFactory
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char <<> at index 37: org/teavm/classlib/java/lang/TString$<clinit>$lambda$_114_0.clas 
        at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
        at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
        at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
        at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
        at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
        at java.base/java.nio.file.Path.resolve(Path.java:515)
        at cc.tweaked.web.builder.TransformingClassLoader.lambda$findUnmappedFile$1(TransformingClassLoader.java:71)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
        at java.base/java.util.AbstractList$RandomAccessSpliterator.tryAdvance(AbstractList.java:706)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
        at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
        at cc.tweaked.web.builder.TransformingClassLoader.findUnmappedFile(TransformingClassLoader.java:71)
        at cc.tweaked.web.builder.TransformingClassLoader.findFile(TransformingClassLoader.java:76)
        at cc.tweaked.web.builder.TransformingClassLoader.findResource(TransformingClassLoader.java:131)
        at java.base/java.lang.ClassLoader.getResource(ClassLoader.java:1403)
        at org.teavm.parsing.resource.ClasspathResourceReader.hasResource(ClasspathResourceReader.java:35)
        at org.teavm.parsing.resource.ResourceClassHolderMapper.apply(ResourceClassHolderMapper.java:40)
        at org.teavm.parsing.resource.ResourceClassHolderMapper.apply(ResourceClassHolderMapper.java:27)
        at org.teavm.parsing.ClasspathResourceMapper.apply(ClasspathResourceMapper.java:121)
        at org.teavm.parsing.ClasspathResourceMapper.apply(ClasspathResourceMapper.java:43)
        at org.teavm.common.CachedFunction.apply(CachedFunction.java:51)
        at org.teavm.parsing.resource.MapperClassHolderSource.get(MapperClassHolderSource.java:32)
        at org.teavm.parsing.ClasspathClassHolderSource.get(ClasspathClassHolderSource.java:43)
        at org.teavm.model.PreOptimizingClassHolderSource.optimize(PreOptimizingClassHolderSource.java:48)
        at org.teavm.model.PreOptimizingClassHolderSource.get(PreOptimizingClassHolderSource.java:37)
        at org.teavm.model.PreOptimizingClassHolderSource.get(PreOptimizingClassHolderSource.java:25)
        at org.teavm.dependency.DependencyClassSource.findClass(DependencyClassSource.java:110)
        at org.teavm.dependency.DependencyClassSource.findAndTransformClass(DependencyClassSource.java:88)
        at org.teavm.dependency.DependencyClassSource.lambda$get$0(DependencyClassSource.java:68)
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1220)
        at org.teavm.dependency.DependencyClassSource.get(DependencyClassSource.java:68)
        at org.teavm.dependency.DependencyClassSource.get(DependencyClassSource.java:38)
        at org.teavm.model.emit.ValueEmitter.invoke(ValueEmitter.java:473)
        at org.teavm.model.emit.ValueEmitter.invoke(ValueEmitter.java:497)
        at org.teavm.model.emit.ValueEmitter.invokeSpecial(ValueEmitter.java:509)
        at org.teavm.classlib.impl.lambda.LambdaMetafactorySubstitutor.createBridge(LambdaMetafactorySubstitutor.java:361)
        at org.teavm.classlib.impl.lambda.LambdaMetafactorySubstitutor.substitute(LambdaMetafactorySubstitutor.java:94)
        at org.teavm.dependency.DependencyAnalyzer.processInvokeDynamic(DependencyAnalyzer.java:917)
        at org.teavm.dependency.DependencyAnalyzer.lambda$scheduleMethodAnalysis$7(DependencyAnalyzer.java:492)
        at org.teavm.dependency.DependencyAnalyzer.processQueue(DependencyAnalyzer.java:652)
        at org.teavm.dependency.DependencyAnalyzer.processDependencies(DependencyAnalyzer.java:679)
        at org.teavm.vm.TeaVM.build(TeaVM.java:392)
        at org.teavm.tooling.TeaVMTool.generate(TeaVMTool.java:475)
        at cc.tweaked.web.builder.Builder.buildClasses(Builder.java:83)
        at cc.tweaked.web.builder.Builder.main(Builder.java:42)

> Task :web:compileTeaVM FAILED

FAILURE: Build failed with an exception.                                                                                                                      

* What went wrong:
Execution failed for task ':web:compileTeaVM'.
> Process 'command 'C:\Program Files\Java\jdk-17\bin\java.exe'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.7/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.    

BUILD FAILED in 1m 20s
84 actionable tasks: 13 executed, 71 up-to-date

**Reply by SquidDev** [reply, marked as answer]
That's helpful, thank you! Looks like there were a couple of issues with building the website in Windows. Those should be fixed as of 925092add3560bd765eb0c156c692ff7134330eb, so pulling in the latest commit should fix that.

**Reply by NevesAngelesLeo** [reply]
Thanks it worked! thank you for the help
